data_structures/src/chain.rs

```diff
@@ -2558,16 +2600,31 @@ impl AltKeys {
     /// Insert all bn256 public keys that appeared in the block (from miner and/or committers)
     pub fn insert_keys_from_block(&mut self, block: &Block) {
         // Add miner bn256 public keys
-        if let Some(value) = block.block_header.bn256_public_key.clone() {
-            self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+        if let Some(value) = block.block_header.bn256_public_key.as_ref() {
+            self.insert_bn256(block.block_header.proof.proof.pkh(), value.clone());
         }
         // Add bn256 public keys from commitment transactions
         for commit in &block.txns.commit_txns {
-            if let Some(value) = commit.body.bn256_public_key.clone() {
-                self.insert_bn256(commit.body.proof.proof.pkh(), value);
+            if let Some(value) = commit.body.bn256_public_key.as_ref() {
+                self.insert_bn256(commit.body.proof.proof.pkh(), value.clone());
             }
         }
     }
+
+    pub fn get_rep_ordered_bn256_list(
+        &self,
+        reputation_set: &TotalReputationSet<PublicKeyHash, Reputation, Alpha>,
+    ) -> Vec<Bn256PublicKey> {
+        let pkhs: Vec<PublicKeyHash> = self.bn256.iter().map(|(pkh, _)| *pkh).collect();
+
+        let ordered_alts: Vec<_> = pkhs
+            .into_iter()
+            .sorted_by_key(|&pkh| (reputation_set.get(&pkh), pkh))
```

Comment:
Now you don't need this collect, you can combine the two iterators into one:

```suggestion
        self.bn256.iter().map(|(pkh, _)| *pkh)
            .sorted_by_key(|&pkh| (reputation_set.get(&pkh), pkh)) 
```